### PR TITLE
dockershim: Implement more functions.

### PR DESCRIPTION
**Pull request by yujuhong**
Based on #29553. Only the last two commits are new.

### CODE REVIEW

File: pkg/kubelet/dockershim/docker_container.go [hunk 1]
```diff
@@ -182,14 +185,112 @@ func (ds *dockerService) RemoveContainer(rawContainerID string) error {
...
+	if r.State.Running {
+		// Container is running.
+		state = runtimeApi.ContainerState_RUNNING
+	} else {
+		// Container is *not* running. We need to get more details.
+		//    * Case 1: container has run and exited with non-zero finishedAt
+		//              time.
+		//    * Case 2: container has failed to start; it has a zero finishedAt
+		//              time, but a non-zero exit code.
+		//    * Case 3: Container has been created, but not started (yet).
```
Comment (Random-Liu) [inline pkg/kubelet/dockershim/docker_container.go]: s/Container/container

File: pkg/kubelet/dockershim/docker_container.go [hunk 2]
```diff
@@ -182,14 +185,112 @@ func (ds *dockerService) RemoveContainer(rawContainerID string) error {
...
+				// TODO: consider exposing OOMKilled via the runtimeAPI.
+				// Note: if an application handles OOMKilled gracefully, the
+				// exit code could be zero.
+				reason = "OOMKilled"
+			case r.State.ExitCode == 0:
+				reason = "Completed"
+			default:
+				reason = fmt.Sprintf("Error: %s", r.State.Error)
+			}
+
```
Comment (Random-Liu) [inline pkg/kubelet/dockershim/docker_container.go]: Empty line.

File: pkg/kubelet/dockershim/docker_container.go [hunk 3]
```diff
@@ -182,14 +185,112 @@ func (ds *dockerService) RemoveContainer(rawContainerID string) error {
...
+		} else if !finishedAt.IsZero() && r.State.ExitCode != 0 { // Case 2
+			state = runtimeApi.ContainerState_EXITED
+			// Adjust finshedAt and createdAt time to startedAt time to avoid
```
Comment (Random-Liu) [inline pkg/kubelet/dockershim/docker_container.go]: Apply the original reason? `ErrContainerCannotRun`

**Comment by Random-Liu** [comment]
LGTM with nits.

File: pkg/kubelet/dockershim/docker_container.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_container.go]: done

File: pkg/kubelet/dockershim/docker_container.go (hunk 3 above)
Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_container.go]: done

**Comment by yujuhong** [comment]
Addressed the comments and rebased on to the latest commit of #29553.
This PR is relatively small compared with others.

**Comment by yujuhong** [comment]
Rebased.

**Comment by yujuhong** [comment]
@k8s-bot test this please issue: #26760

**Comment by dchen1107** [comment]
LGTM

**Pull request merged**